### ISSUE TITLE
[docs] _undirected_edge_index docstring claims self-loops are added, but none are

### ISSUE THREAD

**Issue by PeterPonyu** [PeterPonyu/niche-lens-st#78]
## Problem
The docstring of `_undirected_edge_index` states that self-loops are added so isolated/center nodes always aggregate their own feature during message passing. The function does no such thing — it only symmetrizes the edge list. No self-loop is ever appended. The own-feature is actually retained via the GraphSAGE-style `concat([h, agg])` in `_MeanAggLayer`, not via self-loops, so the docstring describes a mechanism that does not exist.

## Evidence
- `src/nichelens_st/encoder.py:75-85` — docstring promises self-loops, code only symmetrizes:
```python
def _undirected_edge_index(edges: np.ndarray) -> "Tensor":
    """Symmetrize a (2, n_edges) int64 COO array into a torch edge index.

    Self-loops are added so isolated/center nodes always aggregate their own
    feature during message passing.
    """
    src = edges[0]
    dst = edges[1]
    sym_src = np.concatenate([src, dst])
    sym_dst = np.concatenate([dst, src])
    return torch.from_numpy(np.stack([sym_src, sym_dst]).astype(np.int64))
```
- `src/nichelens_st/encoder.py:97-106` — own feature comes from the concat, not self-loops (`agg` is the *neighbor* mean; for an isolated node `deg.clamp_min(1.0)` makes `agg == 0`):
```python
agg = agg / deg.clamp_min(1.0)
return self.lin(torch.cat([h, agg], dim=1))
```

## Why it matters
Misleading docstring for a core message-passing primitive: a maintainer reading it may assume self-loops are present (e.g. when reasoning about degree normalization or porting the layer), leading to incorrect changes. The claim contradicts the actual receptive-field/normalization behavior.

## Suggested fix
- Correct the docstring to state only that the edge list is symmetrized, and that each node's own feature is preserved through the `[h ; mean(neighbors)]` concatenation (no self-loops are added). Alternatively, if self-loops are actually desired, add them explicitly and adjust the degree denominator.

## Acceptance criteria
- [ ] `_undirected_edge_index` docstring accurately describes its behavior (symmetrize only; no self-loops), or self-loops are actually added.

**Dimension:** code-quality/docs
**Severity:** low
